### PR TITLE
github-ci: add more cleanup for running WS

### PR DESCRIPTION
**Pull request by avtikhon**
gitub-ci: add more cleanup for running WS

Found that 'actions/checkout' does not remove all temporary files from
previous runs in submodules [1], it runs only 'git clean -xffd' [2]:
```
  libgcov profiling error:/home/ubuntu/actions-runner/_work/tarantool/tarantool/src/lib/small/CMakeFiles/small.dir/small/small_class.c.gcda:overwriting an existing profile data with a different timestamp
```
To avoid of it added:
```
  git submodules foreach --recursive 'git clean -xffd'
```
to 'actions/environment' which is run after each 'actions/checkout'.

Part of #5986

[1]: https://github.com/tarantool/tarantool/runs/2318244478?check_suite_focus=true#step:5:4012
[2]: https://github.com/actions/checkout/issues/358

### CODE REVIEW

**Review by cyrillos — APPROVED** [review]
Helped me a lot. Thanks!

**Comment by Totktonada** [comment]
Typo: gitub → github.

**Comment by Totktonada** [comment]
> Part of #5986

Does not it fix the whole problem?

File: .github/actions/environment/action.yml [hunk 1]
```diff
@@ -52,4 +52,7 @@ runs:
         if ${{ ! startsWith(github.ref, 'refs/tags/') }}; then
             git tag -d "$(git tag --points-at HEAD)" 2>/dev/null || true
         fi
+        # Found that action/checkout does not remove all temporary
+        # files from previous runs, to avoid of it need to run:
+        git submodule foreach git clean -xffd
```
Comment (Totktonada) [inline .github/actions/environment/action.yml:57]: AFAIR, quotes around `git clean -xfd` are required on some git versions.

**Review by Totktonada — APPROVED** [review]
LGTM aside minor tweaks.

File: .github/actions/environment/action.yml [hunk 2]
```diff
@@ -52,4 +52,7 @@ runs:
         if ${{ ! startsWith(github.ref, 'refs/tags/') }}; then
             git tag -d "$(git tag --points-at HEAD)" 2>/dev/null || true
         fi
+        # Found that action/checkout does not remove all temporary
```
[the remark this answers is not in the document]
Comment (avtikhon) [reply .github/actions/environment/action.yml]: Right, corrected.

File: .github/actions/environment/action.yml (hunk 1 above)
Comment (avtikhon) [reply .github/actions/environment/action.yml]: Ok, sure, let's add it.

**Comment by avtikhon** [comment]
> > Part of #5986
> 
> Does not it fix the whole problem?

Seems not, I've found how to reproduce it locally. It is possible that we need some cleanup instructions for such situations, but I would suggest left the issue opened to discuss it.

**Review by igormunkin — APPROVED** [review]
@avtikhon, thanks for the fix! LGTM.

**Comment by Totktonada** [comment]
Pushed as 2.8.0-233-g57ab8c2e7, 2.7.1-193-g7513575e4, 2.6.2-187-ge327bd48a, 1.10.9-118-ge7d88cf11.

**Pull request merged**